test/functional/pdo_sqlsrv/pdo_574_next_rowset.phpt

```diff
@@ -0,0 +1,154 @@
+--TEST--
+GitHub issue 574 - Fetch Next Result Test
+--DESCRIPTION--
+Verifies the functionality of PDOStatement nextRowset
+--ENV--
+PHPT_EXEC=true
+--SKIPIF--
+<?php require('skipif_mid-refactor.inc'); ?>
+--FILE--
+<?php
+
+require_once("MsCommon_mid-refactor.inc");
+
+try {
+    $conn = connect();
+    $tableName = 'test574';
+    $tableName1 = 'test574_1';
+
+    // create two tables with max fields
+    $columns = array(new ColumnMeta('varchar(max)', 'col1'));
+    createTable($conn, $tableName, $columns);
+
+    $columns = array(new ColumnMeta('varchar(max)', 'col1'));
+    createTable($conn, $tableName1, $columns);
+
+    // insert one row to each table
+    $phrase = str_repeat('This is a test ', 250);
+    $stmt = insertRow($conn, $tableName, array('col1' => $phrase));
+    unset($stmt);
+
+    $phrase = str_repeat('This is indeed very long ', 300);
+    $stmt = insertRow($conn, $tableName1, array('col1' => $phrase));
+    unset($stmt);
+
+    // run queries in a batch
+    $stmt = $conn->prepare("SELECT * FROM [$tableName]; SELECT artist FROM [cd_info]; SELECT * FROM [$tableName1]");
+    $stmt->execute();
+    
+    // fetch from $tableName
+    $row = $stmt->fetch(PDO::FETCH_NUM);
+    if ($row) {
+        echo(substr($row[0], 0, 15)) . PHP_EOL;
+    }
+
+    // fetch from cd_info
+    echo "1. next result? ";
+    $next = $stmt->nextRowset();
+    var_dump($next);
+
+    $row = $stmt->fetch(PDO::FETCH_NUM);
+    if ($row) {
+        echo(substr($row[0], 0, 15)) . PHP_EOL;
+    }
+
+    // fetch from $tableName1
+    echo "2. next result? ";
+    $next = $stmt->nextRowset();
+    var_dump($next);
+
+    $row = $stmt->fetch(PDO::FETCH_NUM);
+    if ($row) {
+        echo(substr($row[0], 0, 25)) . PHP_EOL;
+    }
+
+    // should be no more next results, first returns false second throws an exception
+    echo "3. next result? ";
+    $next = $stmt->nextRowset();
+    var_dump($next);
+
+    $row = $stmt->fetch(PDO::FETCH_NUM);
+    if ($row) {
+        fatalError("This is unexpected!\n");
+    }
+
+    echo "4. next result? " . PHP_EOL;
+    try {
+        $next = $stmt->nextRowset();
+    } catch (PDOException $e) {
+        echo $e->getMessage() . PHP_EOL;
```

Comment:
This is a regression in 5.1.1-preview (see https://github.com/Microsoft/msphpsql/issues/581). newRowset() should not return any error. Please use 5.1.0-preview for testing and see what the behavior should be. Oh.. that means your test will fail against the dev branch for now... maybe talk to David to see when he'll revert the change and merge it with dev?